app/src/main/java/com/example/cocktails/ui/main/ARFragment.kt

```diff
@@ -4,54 +4,71 @@ import android.app.Activity
 import android.app.ActivityManager
 import android.content.Context
 import android.content.Context.ACTIVITY_SERVICE
+import android.net.Uri
 import android.os.Build
 import android.os.Bundle
 import android.util.Log
+import android.view.Gravity
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import android.widget.TextView
 import android.widget.Toast
+import androidx.annotation.RequiresApi
 import androidx.fragment.app.Fragment
-import androidx.lifecycle.Observer
-import androidx.lifecycle.ViewModelProvider
-import com.example.cocktails.Cocktail
-import com.example.cocktails.Cocktails
 import com.example.cocktails.R
-import com.example.cocktails.ScrollingActivity
+import com.google.ar.sceneform.rendering.ModelRenderable
+import com.google.ar.sceneform.ux.ArFragment
+import java.util.function.Consumer
 
 
 class ARFragment(context: Context) : Fragment() {
 
     private var TAG: String = "COMPATIBILITY"
     private var MIN_OPENGL_VERSION: Double = 3.0
 
-//    private lateinit var pageViewModel: PageViewModel
+    var arFragment: ArFragment? = null
+    var glassRenderable: ModelRenderable? = null
 
+    @RequiresApi(Build.VERSION_CODES.N)
+    @Override
+//    @SuppressWarnings({"AndroidApiChecker", "FutureReturnValueIgnored"})
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-//        pageViewModel = ViewModelProvider(this).get(PageViewModel::class.java).
-//        apply {setIndex(arguments?.getInt(PlaceholderFragment.ARG_SECTION_NUMBER) ?: 1)
-//        }
     }
 
+    @RequiresApi(Build.VERSION_CODES.N)
+    @Override
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
 
+        val activity: Activity = requireActivity()
+
         // Check compatibility to AR. if none, just return.
-        if (!checkCompatibility(requireActivity())) {
+        if (!checkCompatibility(activity)) {
             return super.onCreateView(inflater, container, savedInstanceState)
         }
 
         // Create the AR layout and return it
         val root = inflater.inflate(R.layout.ar_layout, container, false)
-        val textView: TextView = root.findViewById(R.id.section_label)
-//        pageViewModel.text.observe(viewLifecycleOwner, Observer<String> {
-//            textView.text = it
-//        })
+
+        arFragment = requireActivity().supportFragmentManager.findFragmentById(R.id.ux_fragment)
+                as ArFragment
+        ModelRenderable.builder()
+            .setSource(activity, Uri.parse("cocktail_glass.sfb"))
+            .build()
+            .thenAccept(Consumer { renderable: ModelRenderable ->
```

Comment:
@shahafhermann is it a long process? user might experience some delay? I want to add a splash screen for cases like that, let me know if it might become relevant to you as well, and I will add some cache for you as well (during the splash screen we will load all heavy data to a local cache, and application will be much faster)

Reply:
Just tried it, it's pretty fast. by the time the user sets up their AR it's already built...